internal/parser/translate_httproute_test.go

```diff
@@ -0,0 +1,445 @@
+package parser
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	gatewayv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	"github.com/kong/go-kong/kong"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/kongstate"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/util"
+)
+
+func Test_fromHTTPRoutes(t *testing.T) {
+	httpPort := gatewayv1alpha2.PortNumber(80)
+	pathMatchPrefix := gatewayv1alpha2.PathMatchPathPrefix
+	pathMatchRegex := gatewayv1alpha2.PathMatchRegularExpression
+	queryMatchExact := gatewayv1alpha2.QueryParamMatchExact
+
+	for _, tt := range []struct {
+		msg      string
+		routes   []*gatewayv1alpha2.HTTPRoute
+		expected ingressRules
+		errMsgs  []string
+	}{
+		{
+			msg:    "an empty list of HTTPRoutes should produce no ingress rules",
+			routes: make([]*gatewayv1alpha2.HTTPRoute, 0),
```

Comment:
any good reason not to use `nil` for an empty slice?

Reply:
nope just vim auto-completions which I probably need to disable. removed in df6143bd400d21d5f7b9178067220d0e3ed5d903